### PR TITLE
Reader: add Daily Post button

### PR DESCRIPTION
**Pull request by jhnstn**
This adds a button to Daily Post posts that pre-fills the editor with the pingback url for the challenge.

### Testing

``` npm run test-client client/reader/daily-post ```

### Screens

---

**Full post view**

screen shot 2016-08-04 at 11 27 45 am

---

**Stream view**
screen shot 2016-08-04 at 11 27 54 am

---

**Pre-filled editor**

screen shot 2016-08-03 at 5 47 18 pm
 *note*: focus is on the title 

cc @blowery 

Test live: https://calypso.live/?branch=add/reader/daily-post-prompt

### CODE REVIEW

**Comment by jhnstn** [comment]
@jancavan @fraying I'm looking for feedback / suggestions on the 'start your post' button placement and copy. 

Also, not sure if we want to add the call to action on the stream card.

**Comment by blowery** [comment]
> Also, not sure if we want to add the call to action on the stream card

I think we do. Maybe a primary button too, so it has some color. 

This should push a new stat to MC and tracks too.

**Comment by jhnstn** [comment]
> This should push a new stat to MC and tracks too.
That was next on my list

I'll give primary buttons a wirl

**Comment by fraying** [comment]
First: I love this.

Various Thoughts:

- I think it would be appropriate, because clicking the button initiates a New Post, for us to pull in a small version of the New Post icon to use in this button. 

screen shot 2016-08-04 at 10 11 18 am

- I also think that the text in the button should be Title Case (not ALLCAPS) to better match other buttons/links in Reader.

- As for the text, could we pull in the prompt? If so, the text in this case could be "Post About Muse". Or maybe "Write about Muse"?

- As for placement, I think left-aligning the button would be better. It's a more direct path for the eye to go from the prompt text to the button, and in the stream view, it would place the button above the "visit" link, which makes sense. 

That's all I got. I think this is fabulous and will be a welcome addition to the "close the read-write loop" goal.

**Comment by jhnstn** [comment]
"Post about xyz" is probably the better option since writing is just one way to participate; There are photo challenges. 

But we could switch it up per type, "Write about xyz" for daily prompts and "Post about abc" for photo challenges.

**Comment by jancavan** [comment]
A non-challenge post returns "undefined" so perhaps those ones shouldn't contain a button:

screenshot 2016-08-04 10 47 16

screenshot 2016-08-04 10 46 44

**Comment by jhnstn** [comment]
@jancavan fixed non-challenge posts in https://github.com/Automattic/wp-calypso/pull/7274/commits/c701204d13ab1882d4a3515488d7f4709389bec7

**Comment by blowery** [comment]
Ideally I think we'd alter the "via ..." text too to make it clearer why that link is there. Maybe "An entry in Community Pool" or "My response to Love". If possible. Might need to teach the editor a new way to accept that footer text to pull it off.

**Comment by blowery** [comment]
Also, this is awesome! So happy to see this moving forward!

**Comment by jhnstn** [comment]
Yeah I opened up the 'via' options to the editorial team with "this is what we can do as is" but totally open to updating the editor to accommodate something better.  

Also, right now the title is linked to the via anchor so it gets tricky to use a nice default title yet keep the via link relevant . 

I was thinking of keeping the title short like "Muse" but then the pingback could say: "My response to: Muse"

**Comment by jancavan** [comment]
Another thing, the entire div (in red) is clickable and pulls up the popover:

screenshot 2016-08-04 11 04 09

A little margin of around `15px` in between the button and the last paragraph would be nice too.

**Comment by jhnstn** [comment]
@jancavan fixed clickable div issue in https://github.com/Automattic/wp-calypso/pull/7274/commits/f103df9f9203a38adfedfc07e294a37b4432f5e0

**Comment by jhnstn** [comment]
Here are the latest screenshots:

stream: 
screen shot 2016-08-04 at 11 27 54 am

full-post:
screen shot 2016-08-04 at 11 27 45 am

**Comment by fraying** [comment]
Wow! Go team! I love this. 👏

**Comment by jancavan** [comment]
@jhnstn That looks good!

**Comment by blowery** [comment]
So what's left? Pushing the stats and the page view?

**Comment by jhnstn** [comment]
@blowery yup, just wrapping it up

File: client/reader/daily-post/index.jsx [hunk 1]
```diff
@@ -90,13 +91,21 @@ class DailyPostButton extends React.Component {
 	}
 
 	pickSiteToPostTo( siteSlug ) {
+		recordAction( 'daily_post_challenge' );
+		recordGaEvent( 'Clicked on Daily Post challenge' );
+		recordTrack( 'calypso_reader_daily_post_challenge_to_site' );
```
Comment (blowery) [inline client/reader/daily-post/index.jsx]: maybe name this `calypso_reader_daily_post_challenge_site_picked` to keep with the tracks naming convention?

Comment (blowery) [inline client/reader/daily-post/index.jsx]: and we should use `recordTrackForPost` here to pick up all of the post attributes we like to report

File: client/reader/daily-post/index.jsx [hunk 2]
```diff
@@ -90,13 +91,21 @@ class DailyPostButton extends React.Component {
...
 		page( `/post/${ siteSlug }?${ qs.stringify( pingbackAttributes ) }` );
 		return true;
 	}
 
 	toggle( event ) {
 		preventDefault( event );
+		if ( ! this.state.showingMenu ) {
+			recordAction( 'open_daily_post_challenge' );
+			recordGaEvent( 'Opened Daily Post Challenge' );
+			recordTrack( 'calypso_reader_daily_post_challenge_opened' );
```
Comment (blowery) [inline client/reader/daily-post/index.jsx]: same here, prefer `recordTrackForPost`

**Comment by blowery** [comment]
ready, assuming tests pass :)

**Comment by jhnstn** [comment]
ok @blowery last bit added.

File: client/reader/daily-post/index.jsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import classnames from 'classnames';
+import page from 'page';
+import qs from 'qs';
+import { get, defer, noop } from 'lodash';
```
Comment (blowery) [inline client/reader/daily-post/index.jsx]: why the `noop`?

**Comment by blowery** [comment]
Looks good, with or without the `noop`.

File: client/reader/daily-post/index.jsx (hunk 3 above)
Comment (jhnstn) [inline client/reader/daily-post/index.jsx]: that got left in after sketching out the `markPostSeen` bit. just removed it.

File: client/reader/daily-post/index.jsx [hunk 4]
```diff
@@ -91,10 +94,14 @@ class DailyPostButton extends React.Component {
...
 
 	pickSiteToPostTo( siteSlug ) {
+		const pingbackAttributes = getPingbackAttributes( this.props.post );
+
 		recordAction( 'daily_post_challenge' );
 		recordGaEvent( 'Clicked on Daily Post challenge' );
 		recordTrackForPost( 'calypso_reader_daily_post_challenge_site_picked', this.props.post );
-		const pingbackAttributes = getPingbackAttributes( this.props.post );
+
+		markPostSeen( this.props.post );
```
Comment (jhnstn) [inline client/reader/daily-post/index.jsx]: Adding this to record a post view if the user clicks the "Post About Abc" from a feed.

**Pull request merged**